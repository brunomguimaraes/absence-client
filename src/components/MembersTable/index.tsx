
import React, { useEffect, useState } from 'react';
import api from '../../services/api';

interface Member {
  id: number,
  name: string,
  crewId: number,
  image: string,
  userId: number,
}

const MembersTable = () => {
  const [members, setMembers] = useState<Member[]>([]);

  useEffect(() => {
    api.get('members').then(response => {
      setMembers(response.data);
    })
  }, []);

  return (
    <div className={"MembersTable"}>
      <h1>User Table</h1>
      <table>
        <tr>
          <th>Name</th>
          <th>Image Url</th>
          <th>Crew ID</th>
          <th>User ID</th>
        </tr>
        {members.map(member => (
          <tr key={member.id}>
            <td>{member.name}</td>
            <td>{member.image}</td>
            <td>{member.crewId}</td>
            <td>{member.userId}</td>
          </tr>
        ))}
      </table>
    </div>
  );
};

export default MembersTable;
