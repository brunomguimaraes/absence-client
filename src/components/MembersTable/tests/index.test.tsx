import React from 'react';
import { shallow } from 'enzyme';
import MembersTable from '..';

describe('MembersTable', () => {
  it('renders the component', () => {
    const component = shallow(<MembersTable />);
    expect(component).toMatchSnapshot();
  });
});

