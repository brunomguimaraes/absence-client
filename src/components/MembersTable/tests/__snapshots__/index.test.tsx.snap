// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersTable renders the component 1`] = `ShallowWrapper {}`;
